### ISSUE TITLE
Last Epoch (899770)

### ISSUE THREAD

**Issue by HolySoap** [ValveSoftware/Proton#6642]
# Compatibility Report
- Name of the game with compatibility issues: Last Epoch
- Steam AppID of the game: 899770

## System Information
- GPU: AMD RX 5700 XT
- Driver/LLVM version: Mesa 23.0.0 / LLVM 15.0.7
- Kernel version: 6.2.6-1-default
- Link to full system information report as Gist: https://gist.github.com/HolySoap/e011e8314e993a82ba9a9ccd15547bab
- Proton version: 6.3-8, 7.0-6, Experimental, GE 7-51 + 7-53

## I confirm:
- [ X ] that I haven't found an existing compatibility report for this game.
- [ X ] that I have checked whether there are updates for my system available.

 
[steam-899770.tar.gz](https://github.com/ValveSoftware/Proton/files/11060322/steam-899770.tar.gz)

## Symptoms 
The GPU gets maxed out and creates a bottleneck, changing graphical settings changes almost nothing and stays between ~12-27fps on Very High preset and gains +10fps on Very Low.

## Reproduction
Simply starting the game is enough, the horrible performance begins in the login screen.

**Comment by HolySoap** [comment]
The reason why I created that issue for a Linux native game is that the native version is completely broken for fours years now (the start of Early Access) and the developers are unable/unwilling to fix this version. It gets more broken with every update. ProtonDB should tell a story in what shape this game is.

**Comment by joghert** [comment]
I have a: AMD R5 5600X, RX 6600XT, 32GB RAM

At 1440p I also get very low fps, when i switch to 1080p it is 60fps in most areas, but in some like deserts it can drop to 25. The quality settings seem to matter less then the resolution.

**Comment by HolySoap** [comment]
Now tested with Proton 8.0-1 and still no change.

**Comment by tethragon** [comment]
I confirm the problem. I have a 5800X3D + Radeon 5600XT. I play the game with native client where I get 100-250 FPS (1080p medium settings). However, native client has some graphical glitches (flashes in some maps and mobs) that could cause epilepsy lol.... So I tried to play through proton. No matter the proton version I tried, no matter the hundreds of tinkering options, I could get max 30 FPS. The problem seems to appear for Radeon owners. Nvidia owners report that they can play through proton nicely. I updated to latest mesa drivers, but there was no change.

**Comment by tethragon** [comment]
Tried today latest ProtonGE 8.4, no change.
Tried latest proton experimental, no change.

Even if the problem will never be fixed, I would love to know what the problem is. I can play through proton countless games, with similar (or better) to Windows performance. Last Epoch? Hell no.....

My system specs:

george@ryzen:~$ inxi -SPARM -GCDN -v1 -xGCRS
System:
  Host: ryzen Kernel: 6.3.7-060307-generic x86_64 bits: 64 compiler: N/A
    Desktop: Cinnamon 5.6.8 Distro: Linux Mint 21.1 Vera
    base: Ubuntu 22.04 jammy
Machine:
  Type: Desktop Mobo: ASUSTeK model: ROG STRIX B350-F GAMING v: Rev X.0x
    serial: <superuser required> UEFI: American Megatrends v: 6042
    date: 04/28/2022
CPU:
  Info: 8-core model: AMD Ryzen 7 5800X3D bits: 64 type: MT MCP arch: Zen 3
    rev: 2 cache: L1: 512 KiB L2: 4 MiB L3: 96 MiB
  Speed (MHz): avg: 2200 min/max: 2200/4549 boost: enabled cores: 1: 2200
    2: 2200 3: 2200 4: 2200 5: 2200 6: 2200 7: 2200 8: 2200 9: 2200 10: 2200
    11: 2200 12: 2200 13: 2200 14: 2200 15: 2200 16: 2200 bogomips: 108588
  Flags: avx avx2 ht lm nx pae sse sse2 sse3 sse4_1 sse4_2 sse4a ssse3 svm
Graphics:
  Device-1: AMD Navi 10 [Radeon RX 5600 OEM/5600 XT / 5700/5700 XT]
    vendor: Sapphire driver: amdgpu v: kernel bus-ID: 09:00.0
  Display: x11 server: X.Org v: 1.21.1.4 driver: X: loaded: amdgpu,ati
    unloaded: fbdev,modesetting,radeon,vesa gpu: amdgpu resolution: 1920x1080
  OpenGL: renderer: AMD Radeon RX 5600 XT (navi10 LLVM 15.0.7 DRM 3.52
  6.3.7-060307-generic)
    v: 4.6 Mesa 23.2.0-devel (git-96cf453 2023-06-08 jammy-oibaf-ppa)
    direct render: Yes
Audio:
  Device-1: AMD Navi 10 HDMI Audio driver: snd_hda_intel v: kernel
    bus-ID: 09:00.1
  Device-2: AMD Starship/Matisse HD Audio vendor: ASUSTeK
    driver: snd_hda_intel v: kernel bus-ID: 0b:00.4
  Sound Server-1: ALSA v: k6.3.7-060307-generic running: yes
  Sound Server-2: JACK v: 1.9.20 running: no
  Sound Server-3: PulseAudio v: 15.99.1 running: yes
  Sound Server-4: PipeWire v: 0.3.48 running: yes
Network:
  Device-1: Intel I211 Gigabit Network vendor: ASUSTeK driver: igb v: kernel
    port: e000 bus-ID: 04:00.0
RAID:
  Message: No RAID data found.
Drives:
  Local Storage: total: 7.28 TiB used: 1.52 TiB (20.9%)
  ID-1: /dev/nvme0n1 vendor: Samsung model: SSD 970 EVO Plus 1TB
    size: 931.51 GiB temp: 35.9 C
  ID-2: /dev/sda vendor: Crucial model: CT1000MX500SSD1 size: 931.51 GiB
  ID-3: /dev/sdb vendor: Toshiba model: HDWD130 size: 2.73 TiB
  ID-4: /dev/sdc vendor: Toshiba model: HDWD130 size: 2.73 TiB
Partition:
  ID-1: / size: 313.91 GiB used: 53.96 GiB (17.2%) fs: ext4
    dev: /dev/nvme0n1p4
  ID-2: /boot/efi size: 96 MiB used: 48.5 MiB (50.6%) fs: vfat
    dev: /dev/nvme0n1p3
Info:
  Processes: 416 Uptime: 1h 54m Memory: 31.24 GiB used: 5.01 GiB (16.0%)
  Init: systemd runlevel: 5 Compilers: gcc: 12.1.0 Packages: 3033 Shell: Bash
  v: 5.1.16 inxi: 3.3.13

**Comment by xAcid9** [comment]
CPU: Intel i7-3770
GPU: GTX 980 Ti
Driver version: 530.41.03
Kernel version: 6.3.8
Proton version: Proton 7.0-6, Proton 8.0-2, Experimental, Proton GE 7-37, Proton GE 7-55

Similar issue as OP. Basically it max out VRAM when launching and causing performance + stability issue. Memory leaked? 

Native version is playable but with texture and performance issue.

**Comment by tethragon** [comment]
This is the first game that I get that HUGE difference between windows and Linux+proton. I get 250 FPS in windows and max 30FPS in Linux with proton. I also get 250 fps with native linux client, but the graphical glitches make the game intolerable. Since devs are not going to fix the native linux version (in my opinion), proton is the only option. If only valve devs could try and see what's wrong with the game and performs so horrible.... Maybe something to do with Unity Engine? I don't know....

**Comment by tethragon** [comment]
I am using Linux Mint (x.Org) and I just tested with a more modern display server (fedora 38 kde + wayland). Same results. Awful performance.

**Comment by xAcid9** [comment]
Running the game with OpenGL instead of Vulkan (PROTON_USE_WINED3D) give me better overall performance but still only around 50% of my 980 TI capability. 

VRAM usage was around 3GB instead of maxing out right of the gate with Vulkan.

**Comment by tethragon** [comment]
UPDATE:

I noticed that resizable bar had been somehow deactivated from BIOS. I re-enabled it and this made a huge difference. With proton experimental the performance jumped from 30-40 FPS (1080p medium) to ~100-120 FPS in the starting screen of a monolith. This is still 50% of the performance I get in Windows ~200-250 FPS. Of course in battle with many enemies, the performance is much lower. 

My system is 5800X3D + Radeon 5600XT
TLDR:
1) Enabling resizable bar made a huge difference, more than 150%. This is not normal as resizable bar is not giving that much of a performance boost. So it must be something else, that "enabled" proton to improve performance.
2) Enabling resizable bar, also disables CSM in BIOS. Could this make a difference or explain the awful performance with proton in this game? (for AMD hardware) ?
3) Still the performance is not even close to Windows client performance running in Windows. There must be something wrong that needs to be discovered so Proton could be patched for this game.

**Comment by tethragon** [comment]
Tried again with latest proton release (7-7-2023). Getting at best 50% of the performance I can have in Windows. This game needs some love from proton developers.... I wish a proton dev see this thread and fix the problem. The game has a very healthy playerbase and it's a pity we can't enjoy it using proton (unless you have a very expensive high end gpu).

**Comment by tethragon** [comment]
RDNA-1 GPU owners REJOICE!!!!!!!

After many many many long hours of exhaustive testing, trying hundreds of tinkering methods, I found a workaround!!!!

Using `PROTON_USE_WINED3D=1 %command%`

tells proton to translate to OpenGL and NOT vulkan. This did the trick! The game run at **almost 100% the performance I get in Windows!!!!!!!! Hooray!!!!!!!**

The problem was that **I had to use proton 8.0.2** (and NOT latest experimental) + turn off steam overlay. Using experimental had also no performance issue (100% performance) but the screen was too dark, like the brightness was turned down to almost zero.

Also, adding -force-d3d11 after the %command% may or may not improve performance a bit. I am not sure yet and I am too tired to keep on testing.

None of the protonDB reviews recommended this setting. Maybe this setting works for RDNA-1 GPUs (like my own Radeon 5600XT and maybe RDNA-2 and RDNA-3 have no performance reggression using normal translation to Vulkan but, in my case, this option was the solution!

**NOTE that may help proton devs implemet a fix:**
Up to now, I was using proton experimental and getting 50% of the windows os performance (without WINED3D). Tried older proton versions and the performance was worse. I noticed that the VRAM usage was 6GB/6GB and apparently this is causing the problem, because with the use of  `PROTON_USE_WINED3D=1 ` I monitored the VRAM usage and it went down to 5GB/6GB !!!! This means that there is a bug for this game in the proton translation and makes the GPU starve for VRAM. Also tried in another computer (without WINED3D) with a Radeon 6600XT (8GB) and I noticed that the performance was somewhat better (~60-65% of the windows performance) but AGAIN the VRAM usage was 7.9GB/8GB, almost 100%.  So this is a VRAM leak or something like that problem. Somehow this VRAM problem goes away with the WINED3D option. 

----------------------------------------------------------------------------------------------------------------------------------------------------------------

**UPDATE: TOO GOOD TO BE TRUE.**

Unfortunately playing more, I found out that the WINED3D "solution" has [the same graphical glitches that the native client has...](https://www.reddit.com/r/LastEpoch/comments/14vwbos/investigating_last_epochs_map_blinking_mobs/) Back to square 1. Proton needs to be patched so the vram leak is fixed. There is no other way....

**Comment by MantasVaitkunas** [comment]
Tried out proton 7.0.6 with @tethragon suggestion - "PROTON_USE_WINED3D=1 %command%", it works excelent :)

**Comment by abique** [comment]
Played today, with proton experimental and proton 9.0, both leads to unplayable experience, the FPS is too low, and drops over time.

**Comment by Meneldor** [comment]
Game works fine, but sometime it doesn't launch, seems to be running but nothing start.
Closing and relaunching the game usually do the trick.
Does someone experience the same thing?

**Comment by ThatGuy435** [comment]
> Game works fine, but sometime it doesn't launch, seems to be running but nothing start. Closing and relaunching the game usually do the trick. Does someone experience the same thing?

Can confirm, this is also happening to me on Fedora KDE 41 with 5800x3d/7900XT and native Steam with Proton Experimental.
